### PR TITLE
Adding sparse LO

### PR DESCRIPTION
**Pull request by E1337Kat**

### CODE REVIEW

File: src/load_order.ts [hunk 1]
```diff
@@ -681,6 +855,22 @@ const deployAndSerializeNewLoadOrder: VortexWrappedSerializeFunc = (
   const serializedLoadOrder =
     encodeLoadOrder(v2077LoadOrder);
 
+  const maybeRebuiltModsDotJson: TaskEither<Error, string> = pipe(
```
Comment (effs) [inline src/load_order.ts:858]: Think this needs to wait on the deployment we kick off above to complete, right?

The LO disk write was here outside the async deployment as a fallback if Vortex crashes or something - but tbh I'm not sure how useful or needful that is, so if it's easiest to just put the whole chain in the async world, I think it should be fine.

**Review by effs — COMMENTED** [review]
I think we need to chain the tasks here, but looks really good otherwise!

File: src/load_order.ts (hunk 1 above)
Comment (E1337Kat) [reply src/load_order.ts:858]: mhmm, I will take a look at that then

File: src/load_order.ts [hunk 2]
```diff
@@ -665,14 +826,59 @@ const deployAndSerializeNewLoadOrder: VortexWrappedSerializeFunc = (
...
+      tryCatchTE(
+        () =>
+          startREDmodDeployInTheBackgroundWithNotifications(vortexApi, gameDirPath, loID, v2077LoadOrder, vortexLoadOrder),
+        (err) => {
+          vortexApi.log(`error`, `${me}: REDmod deployment ${loID} failed!`, S(err));
+          showInfoNotification(vortexApi, InfoNotification.REDmodDeploymentFailed);
+          throw err;
+        },
+      ),
+      mapTE(({ stdout, stderr: _stderr }) => {
```
Comment (effs) [inline src/load_order.ts:843]: Tiny bit of additional semantic context if you use [`chain*`](https://gcanti.github.io/fp-ts/modules/TaskEither.ts.html#chain-1)

File: src/load_order.ts [hunk 3]
```diff
@@ -665,14 +826,59 @@ const deployAndSerializeNewLoadOrder: VortexWrappedSerializeFunc = (
...
+            vortexApi.log(`error`, `${me}: Error rebuilding mods.json: ${error.message}`);
+            showInvalidLoadOrderFileErrorDialog(vortexApi, loadOrderPathFor(activeProfile, gameDirPath));
+            return Promise.reject(error);
+          }),
+          mapTE((modsDotJsonList) => {
+            vortexApi.log(`info`, `${me}: Successfully rebuilt mods.json...`);
+            return rebuildModsDotJsonLoadOrder(vortexApi, modsDotJsonLoadOrder.mods, toMutableArray(modsDotJsonList));
+          }),
+          mapTE((rebuiltModsDotJsonLoadOrder) => encodeModsDotJsonLoadOrder(rebuiltModsDotJsonLoadOrder)),
+          mapLeftTE((error) => {
```
Comment (effs) [inline src/load_order.ts:860]: Can drop this one

**Review by effs — COMMENTED** [review]
Looks good! Tests read clean too

This reminds me we probably need to add a bit of Exception hierarchy to both pattern match and to carry notification/dialog types etc. Helps consolidate error handling so that you don't need to fuss with it everywhere separately. Same for some kind of generic tap-style effect for logging.. I've been a bit lax with adding helpers :/

**Comment by E1337Kat** [comment]
Need to revamp to include the deployedHash bits which is not an easy task.

**Comment by E1337Kat** [comment]
Gonna toss this in favor of modlist: #365

**Pull request closed without merging**